src/equilibrate_utils.jl

```diff
@@ -0,0 +1,394 @@
+    push!(func_body.args,Expr(:if,:(length(internal___var___polyvector) != 0),Expr(:block)))
+    foreach(i -> push!(func_body.args[1].args[2].args, :($(params[i])=internal___var___polyvector[$i])), 1:length(params))
+    push!(func_body.args,:([]))
+    foreach(poly -> push!(func_body.args[2].args, recursive_replace!(poly,(reactants,:internal___polyvar___x))), deepcopy(f_expr))
+    func_expr = :((;TO___BE___REMOVED=to___be___removed) -> $(deepcopy(func_body)))
+    foreach(i -> push!(func_expr.args[1].args[1].args,Expr(:kw,params[i],:(internal___reaction___network.polyvars_params[$i]))), 1:length(params))
+    push!(func_expr.args[1].args[1].args,Expr(:kw,:internal___var___polyvector,:([])))
+    deleteat!(func_expr.args[1].args[1].args,1)
+    push!(func_expr.args[1].args,:internal___reaction___network)
+    push!(func_expr.args[1].args,Expr(:kw,:internal___polyvar___x,:(internal___reaction___network.polyvars_vars)))
+    push!(func_expr.args[1].args,Expr(:kw,:print_msg,1))
+    return func_expr
+end
+
+#Manages post creation modification to a reaction network with respect to utility needed for equilibrium calculations. Attempts to make an equilibrium polynomial.
+function manage_equilibrium_functionality!(reaction_network::DiffEqBase.AbstractReactionNetwork)
+    fix_parameters(reaction_network,fill(1,length(reaction_network.params)))
+    reaction_network.is_polynomial_system = try
+        fix_parameters(reaction_network,fill(1,length(reaction_network.params)))
+        true
+    catch
+        false
+    end
+    reaction_network.is_polynomial_system && (try fix_parameters(reaction_network) catch end)
+end
+
+
+
+## Functions required for the preparation to use homotopy continuation to find fixed points.
+
+#Calls the internal make_polynomial function to create a polynomial. Used to model creation (will succced unless there is parameters in the exponents). Input include list of parameters to fix. All parameters occuring as exponent must be here (none other is required at this stage). Will insert fixed concentrations if such exists.
+function fix_parameters(reaction_network::DiffEqBase.AbstractReactionNetwork, params=Vector{Number}()::Vector{Number}; kwargs...)
+    check_is_polynomial(reaction_network)
+    reaction_network.equilibratium_polynomial = reaction_network.make_polynomial(reaction_network;internal___var___polyvector=params,kwargs...)
+    !(typeof(reaction_network.equilibratium_polynomial[1])<:Polynomial) && (reaction_network.equilibratium_polynomial = map(pol->pol.num,reaction_network.equilibratium_polynomial))
+    foreach(sym -> reaction_network.equilibratium_polynomial[findfirst(reaction_network.syms.==sym)] = reaction_network.fixed_concentrations[sym], keys(reaction_network.fixed_concentrations))
+end
+
+#In some networks some linear combinations of concentrations remain fixed. This supplies this information directly to the network.
+macro fixed_concentration(reaction_network::Symbol, fixed_conc::Expr...)
```

Comment:
Another suggestion, instead of:

```
@fixed_concentration rn X+Y=xyTot
```

what about

```
@add_constraints rn begin
X + Y = xyTot
Z + W = zwTot
end
```
The idea being that in the future we could build out this macro to support more general network constraints, like quasi-equilibrium constraints (leading to DAE systems). Using `begin` to `end` parallels the `reaction_network` macro too, which maybe makes it a bit easy to remember for users.

This might also make sense to live in `reaction_network.jl` with the idea we could use it more generally in the future.

I agree that in the future we should add auto-detection methods for conservation laws. There does appear to be some literature on this. For example, with just a couple minutes of searching I found [this](https://arxiv.org/abs/1109.1505).

Reply:
Also, does the order of the equation matter here? i.e. could we instead say:
```
xyTot = X + Y
```
which seems more like a definition of `xyTot` to me, so I find it a bit clearer.

Reply:
This type of definition would work.

the begin/end definition would also work. I think that in many places only one fixed state would be given, making the neater one more preferably. But I agree with your argument, especially if there are several such conditions. Could implement both?

Reply:
How about `add_constraint` for adding one (without begin/end) and `add_constraints` with begin/end for adding multiple. 

Then the only question is whether we should also support calling `add_constraint` multiple times (which we probably should).